forge/forge/components/web/search.py

```diff
@@ -137,8 +136,16 @@ def google(self, query: str, num_results: int = 8) -> str | list[str]:
 
         try:
             # Get the Google API key and Custom Search Engine ID from the config file
-            api_key = self.legacy_config.google_api_key
-            custom_search_engine_id = self.legacy_config.google_custom_search_engine_id
+            api_key = (
+                self.config.google_api_key.get_secret_value()
+                if self.config.google_api_key
+                else None
+            )
+            custom_search_engine_id = (
+                self.config.google_custom_search_engine_id.get_secret_value()
+                if self.config.google_custom_search_engine_id
+                else None
+            )
```

Comment:
maybe cleaner:
```suggestion
            # Should be the case if this command is enabled:
            assert self.config.google_api_key
            assert self.config.google_custom_search_engine_id
```
and then just use `self.config.google_api_key` rather than `api_key`

Reply:
asserts get stripped when building for debug (and I think freezing) and thus shouldn't be used outside tests imo

Reply:
I've checked and can confirm that asserts get stripped when running in optimized mode but this is to silence type checker, so I guess that's ok?
However if I use `assert` here then the tests will fail; tests use vcr cassettes and handle `None` api key. I'm leaving it as is for now.

Reply:
I didn't know `assert` gets stripped, that seems weird. In any case, the original code is also invalid because it can pass `api_key = None` and `custom_search_engine_id = None` to the search call. The only reason the type checker doesn't complain about this is because type stubs for the google API client aren't currently installed so that call isn't checked at all.

> However if I use assert here then the tests will fail; tests use vcr cassettes and handle None api key. I'm leaving it as is for now.
Google search requests [are not cached in VCR](https://github.com/Significant-Gravitas/AutoGPT/blob/9f71cd243766b8f1570fd3c8712857af7ffa4257/autogpt/tests/vcr/vcr_filter.py#L9-L13).

Reply:
Right, google search doesn't use vcr but the tests still fail if I use `assert` :( we mock them and call `google` command with `None` api key.
https://github.com/Significant-Gravitas/AutoGPT/blob/9f71cd243766b8f1570fd3c8712857af7ffa4257/autogpt/tests/unit/test_web_search.py#L101